BUG: Remove reset of seeding in the ML EM `fit` method of `Factor`

jmbejara opened the pull request:
- [x] closes #7357
- [x] tests added / passed. 
- [x] code/documentation is well formatted.  
- [x] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

I removed the line of code in the `_fit_ml_em` method of `multivariate.Factor` that resets the RNG seed. Doing so broke 2 tests cases, but these were only broken due to precision error. These are fixed by simply increasing the number of iterations used to estimate the models in the test cases.


<details>


**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

bashtage commented:
I think a more deterministic fix is something like

```
rs = np.random.RadnomState(1234)
rs.normal(...)
```

which should be identical to the current behaviour without touching the singleton state.  A better optional lets a user pass a RandomState which is passed to fit em when not None, and if None the code above is used.

pep8speaks commented:
Hello @jmbejara! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`statsmodels/multivariate/factor.py`](https://github.com/statsmodels/statsmodels/blob/b05087dfddf2293b25f5e26416ec5092c1dc9230/statsmodels/multivariate/factor.py):

> [Line 440:12](https://github.com/statsmodels/statsmodels/blob/b05087dfddf2293b25f5e26416ec5092c1dc9230/statsmodels/multivariate/factor.py#L440): [F821](https://duckduckgo.com/?q=pep8%20F821) undefined name 'random_state'

jmbejara commented:
@bashtage As you suggested, I added the option to pass a `RandomState` instance that defaults to a generator with the same seed as the original implementation. I reverted the changes I made to the tests, since the code now behaves as it did before unless the user supplies a different generator. Hope this is ok. Let me know if I can do anything else! Thanks

bashtage commented on statsmodels/multivariate/factor.py:
```diff
@@ -433,12 +433,15 @@ def nscore(par):
+            rs = np.random.RandomState(3427)
+        else:
+            rs = random_state
+        load = 0.1 * rs.normal(size=(self.k_endog, self.n_factor))
```
Could you switch to standard_normal?  It is faster than normal.

jmbejara replied:
Yep! Just did it.

bashtage commented:
Looks good. Should hve a test (or modify and existing test) to make sure that the singleton RandomState is not changed by calling em.

jmbejara commented:
@bashtage Yep! Just added a short test for it based on the original issue.

bashtage commented on statsmodels/multivariate/tests/test_ml_factor.py:
```diff
@@ -78,12 +78,33 @@ def test_exact_em():
+        factor = Factor(endog=epsilon, n_factor=2, method='ml').fit()
+        return epsilon[0,0]
+
+    with warnings.catch_warnings():
```
I thinkyou need to indent 4 spaces starting here.

bashtage commented on statsmodels/multivariate/tests/test_ml_factor.py:
```diff
@@ -78,12 +78,33 @@ def test_exact_em():
+
+    This test arose from https://github.com/statsmodels/statsmodels/issues/7357
+    """
+
```
alternative would be `initial = np.random.get_state()`
then later
```
final = np.random.get_state()
```
and check that initial was equal to `final`.
 
But your method gets the point across.

jmbejara replied:
I like that better too. Should have looked more into this. Just changed the test to do it this way.
